caldav/server.go

```diff
@@ -24,7 +25,8 @@ type Backend interface {
 // Handler handles CalDAV HTTP requests. It can be used to create a CalDAV
 // server.
 type Handler struct {
-	Backend Backend
+	Backend              Backend
+	UserPrincipalBackend webdav.UserPrincipalBackend
```

Comment:
We could just embed `webdav.UserPrincipalBackend` into `Backend` to avoid dealing with multiple backends.